src/ApacheOrcDotNet.OptimizedReader/Infrastructure/CompressedData.cs

```diff
@@ -38,7 +38,11 @@ public static int Decompress(ReadOnlySpan<byte> inputBuffer, Span<byte> outputBu
             {
                 var compressedChunkLength = OrcCompressedBlock.GetChunkLength(compressionKind, inputBuffer[inputPosition..]);
 
-                outputPosition += OrcCompressedBlock.DecompressBlock(compressionKind, inputBuffer.Slice(inputPosition, compressedChunkLength), outputBuffer[outputPosition..]);
+                var chunkToDecompress = (inputPosition + compressedChunkLength) > inputBuffer.Length
```

Comment:
I'd rather calculate the length using this ternary and then `Slice` either way rather than having different slices for different conditions.

Reply:
Also shouldn't we just throw if `inputPosition + compressedChunkLength > inputBuffer.Length`?  I assume this is a remaining artifact from using less than a full chunk of the second compressed block.

Reply:
Since we are back using two full chunks, this ternary is no longer needed. Removed.